device_backends/NumericAddressedBackend/src/MapFileParser.cpp

```diff
@@ -109,43 +108,37 @@ namespace ChimeraTK {
 
   /********************************************************************************************************************/
 
-  std::pair<bool, std::pair<unsigned int, unsigned int>> MapFileParser::getInterruptData(std::string accessTypeStr) {
+  std::vector<uint32_t> MapFileParser::getInterruptData(std::string accessTypeStr) {
     std::string strToFind("INTERRUPT");
     auto pos = accessTypeStr.find(strToFind);
-    if(pos == std::string::npos) return {false, {0, 0}};
+    if(pos == std::string::npos) return {};
+    std::vector<uint32_t> retVal;
 
-    unsigned int interruptCtrlNumber = 0;
-    unsigned int interruptNumber = 0;
     accessTypeStr.erase(pos, strToFind.length());
 
-    auto delimiterPos = accessTypeStr.find(':');
-    if(delimiterPos != std::string::npos) {
-      std::string interruptCtrlNumberStr = accessTypeStr.substr(0, delimiterPos);
-      std::string interrupNumberStr = accessTypeStr.substr(delimiterPos + 1);
+    size_t delimiterPos;
+    do {
```

Comment:
I think you can do that with fewer string copying by using from_chars and using the old and new index, but that is probably nitpicking

Reply:
I made a comment in the code so it can be improved when this is touched again.